src/python/pants/backend/jvm/tasks/jvm_compile/javac/javac_compile.py

```diff
@@ -136,12 +139,22 @@ def compile(self, ctx, args, classpath, upstream_analysis,
         settings_args = (a.replace('$JAVA_HOME', distribution.home) for a in settings.args)
       javac_cmd.extend(settings_args)
 
-    javac_cmd.extend([
-      '-d', ctx.classes_dir,
-      # TODO: support -release
-      '-source', str(settings.source_level),
-      '-target', str(settings.target_level),
-    ])
+    if self.execution_strategy() == 'hermetic':
+      javac_cmd.extend([
+        # The comment below is how we will add the output directory once it
```

Comment:
Twitter recently enabled `--compile-zinc-use-classpath-jars` internally because modern zinc supports it a bit better. So yea, I think we're about ready to commit, enable that flag by default, and deprecate it.

@benjyw : In your JVM usecases, have been using `--use-classpath-jars`?

Reply:
No, but I don't see why we couldn't. I'll enable it and see what happens.

Reply:
Looks fine - we poke around in the runtime classpath, which already uses z.jar, I believe.